### PR TITLE
MM-36248 Fix GitLab Button Color and Text not reflecting in UI

### PR DESCRIPTION
**Pull request by engineereng**
#### Summary
Fixes GitLab Button and Color config settings not being reflected in the UI

Asked for a review from @mkraft.
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-36248

#### Related Pull Requests
- Has server changes (https://github.com/mattermost/mattermost-server/pull/17848)

#### Screenshots
None

#### Release Note
```release-note
Fixes GitLab Button and Color 
```

### CODE REVIEW

**Review by mkraft — APPROVED** [review]

File: components/login/login_controller/login_controller.jsx [hunk 1]
```diff
@@ -679,20 +681,31 @@ class LoginController extends React.PureComponent {
...
         if (gitlabSigninEnabled) {
+            const buttonStyle = {};
+            if (this.props.gitlabButtonColor) {
+                buttonStyle.backgroundColor = this.props.gitlabButtonColor;
+            }
+            let buttonText = (
+                <FormattedMessage
+                id='login.gitlab'
+                defaultMessage='GitLab'
+            />
```
Comment (BenCookie95) [inline components/login/login_controller/login_controller.jsx:692]: The formatting looks off here, try running `npm run fix`

**Review by BenCookie95 — CHANGES_REQUESTED** [review]
We should probably add a snapshot test to check the configs are getting applied! Let me know if you need help with that!

Also, run `npm run fix` to fix/check your lint

**Comment by BenCookie95** [comment]
@engineereng I don't think you pushed the snapshot updates from the __snapshot__ directory

**Comment by engineereng** [comment]
> @engineereng I don't think you pushed the snapshot updates from the **snapshot** directory

Sorry about that! I pushed them yesterday.

**Review by BenCookie95 — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]
Locally Passed. Working as Expected.
image

image

**Pull request merged**